[Wallet] Request coda modal

o1pranay opened the pull request:
PTAL @Schmavery @figitaki 

Added the modal to select wallet and copy pk.  Styling is not done, but figured would be better to throw it up and get yall's code review first.

<img width="800" alt="" src="https://user-images.githubusercontent.com/50593560/58777522-83cd7180-8583-11e9-9210-22f293013f9e.png">

o1pranay commented on frontend/wallet/src/render/components/Footer.re:
```diff
@@ -79,70 +84,93 @@ module SendPaymentMutation = ReasonApollo.CreateMutation(SendPayment);
+            <Button
+              label="Request"
+              style=Button.Gray
+              onClick={_ => setModalState(_ => Some("active wallet"))}
```
yes - will get rid of this

o1pranay commented on frontend/wallet/src/render/components/Footer.re:
```diff
@@ -79,70 +84,93 @@ module SendPaymentMutation = ReasonApollo.CreateMutation(SendPayment);
+            />
+            {switch (modalState) {
+              | None => React.null
+              | _ =>
```
switched it to a boolean.  Tried passing in the activePublicKey here, but the problem is that the None case swallows both modal being closed, and a wallet not being active.  This prevented the modal from opening when a wallet wasn't selected, which seemed weird

Schmavery replied:
yeah, you'd have to have an `option(option(PublicKey.t))` to express the state in that case I guess

o1pranay commented:
@Schmavery added 2 more things:
- updated some UI elements:
  - wallet list hover in settings
  - top bevel for header row
- removed the fetch-policy since your PR was merged in

Schmavery commented on frontend/wallet/src/render/components/RequestCodaModal.re:
```diff
@@ -0,0 +1,98 @@
+    let (settings, _updateAddressBook) = React.useContext(AddressBookProvider.context);
+    let (selectedWallet, setSelectedWallet) = React.useState(() => activePublicKey);
+    let keyString = switch (selectedWallet) {
+      | None => ""
```
This empty string state never gets used, as far as I can tell. You could probably avoid needing to represent this state by only using it in your switch cases below (instead of matching against `Some(_selectedWallet)`)

Schmavery commented on frontend/wallet/src/render/views/settings/SettingsPage.re:
```diff
@@ -50,19 +49,28 @@ module Styles = {
       style([
         userSelect(`none),
-        padding2(~v=`rem(1.), ~h=`rem(0.25)),
+        padding2(~v=`rem(1.), ~h=`rem(1.)),
```
if they're the same now you can just do padding instead of paddin2

Schmavery commented on frontend/wallet/src/render/views/settings/SettingsPage.re:
```diff
@@ -50,19 +49,28 @@ module Styles = {
+        hover([
+          backgroundColor(Theme.Colors.midnightAlpha(0.05)),
+          selector(
+            "> :last-child",
```
I don't understand this selector

o1pranay replied:
selects the last nested child of the wallet item (the span with chevron)

tried to use the classname .walletChevron in the selector, but couldn't get it to work - if you have another way of accomplishing this, then happy to switch it out
